File: backend/src/Core/Models/CompanyFundamentals/CompanyProfile.cs
namespace Core.Models.CompanyFundamentals;

public class CompanyProfile
{
    public required string Symbol { get; set; }
    public float Price { get; set; }
    public double MarketCap { get; set; }
    public float LastDividend { get; set; }
    public string? Range { get; set; }
    public float Change { get; set; }
    public float ChangePercentage { get; set; }
    public int Volume { get; set; }
    public string? Exchange { get; set; }
    public string? Website { get; set; }
    public string? Image { get; set; }
}

File: client-ng/src/app/shared/models/fundamentals/company-profile.ts
export type CompanyProfile = {
  symbol: string;
  price: number;
  marketCap: number;
  lastDividend: number;
  range: string;
  change: number;
  changePercentage: number;
  volume: number;
  exchange: string;
  website: string;
  image: string;
};


export type KeyMetrics = {
  symbol: string;
  enterpriseValueTtm: number;
  returnOnInvestedCapitalTtm: number;
  currentRatio: number;
  netDebtToEbitdaTtm: number;
};


export type FinancialRatios = {
  symbol: string;
  debtToEquityRatioTtm: number;
  grossProfitMarginTtm: number;
  priceToEarningsRatioTtm: number;
  forwardPriceToEarningsGrowthRatioTtm: number;
  priceToSalesRatioTtm: number;
  priceToBookRatioTtm: number;
  priceToFreeCashFlowRatioTtm: number;
};

